marconi-chain-index/src/Marconi/ChainIndex/Experimental/Indexers/Worker.hs

```diff
@@ -63,7 +84,29 @@ catchupWorkerWithFilter config eventFilter indexer =
   let chainPointDistance :: Core.Point (WithDistance a) -> WithDistance a -> Word64
       chainPointDistance _ = Distance.chainDistance
       mapEventUnderDistance = fmap sequence . traverse (eventExtractor config)
-   in Core.createWorker (workerName config) mapEventUnderDistance $
-        Core.withTrace (logger config) $
-          Core.withCatchup chainPointDistance (catchupConfig config) $
-            Core.withTransform id (eventFilter . Distance.getEvent) indexer
+   in do
+        let getResumablePoints _securityParam ix = do
+              fmap List.singleton $ Core.lastSyncPoint ix
```

Comment:
I initially thought of using a typeclass (`HasResumePoints`, in `WithResume`) and I'm not a huge fan of this design. We could also add a different query to get the `n` last points for any n, and we could remove `HasResumePoints` (so we get rid of an ad-hoc typeclass).

Reply:
By different query, you mean something like `lastSyncPoints :: m [Point event]`?

Reply:
No I was thinking about reusing the `Queryable` interface. I did find an advantage to `Queryable actually: transformers already know how to handle them (otherwise you need to add extra logic for your code to go through the transformers layers).

Reply:
So create a new datatype query for Queryable for getting the last sync points? Why not just extend `IsSync`?

Reply:
Because being able to provide access to the `n` last events is a much stronger requirements than knowing the last sync points.
Later, we may want to add ephemeral indexers, these indexers may want to expose their last sync points but may have no interest in providing a list of their last sync points.